src/main/java/net/sf/jabref/logic/util/strings/StringUtil.java

```diff
@@ -646,7 +646,14 @@ public static String expandAuthorInitials(String name) {
                 }
                 for (int j = 1; j < names.length; j++) {
                     if (j == 1) {
-                        sb.append(Author.addDotIfAbbreviation(names[j]));
+                        String[] splittedNames = names[j].split(" ");
+                        if (splittedNames.length > 0) {
+                            sb.append(Author.addDotIfAbbreviation(splittedNames[0]));
+                        }
+                        for (int k = 1; k < splittedNames.length; k++) {
+                            sb.append(' ');
+                            sb.append(Author.addDotIfAbbreviation(splittedNames[k]));
+                        }
```

Comment:
The StringBuilder has nothing to do with the StringJoiner in that case. Just use this solution:

```java
String innerNames = Arrays.stream(names[j].split(" "))
    .map(Author::addDotIfAbbreviation)
    .collect(Collectors.joining(" "));
sb.append(innerNames);
```